gcc/rtl-rvtt-schedule.c

```diff
@@ -202,10 +222,13 @@ static void dynamic_schedule_wh_bh(rtx_insn *insn)
 // "dynamic" since it depends on the instructions
 static void transform ()
 {
-  DUMP("Schedule pass on: %s\n", function_name(cfun));
+  DUMP ("Schedule pass on: %s\n", function_name(cfun));
 
   basic_block bb;
 
+  // We want to reuse the same vector, to avoid reallocations.
+  std::vector<basic_block> visited;
```

Comment:
Nit : You could reserve the memory for visited. In a pathological case where the number of elements pushed back in different iterations vary in a certain order, you would still face reallocations.

Reply:
sure, it's simple to get the number of BBs -- my intent was only to reduce the overall reallocs.

Reply:
What is the size of a basic block data type ? Is it heavy ? Does it make sense to push back pointers to it in the vector. I assume its light.

Reply:
it's a pointer